plugins/catalog/src/components/AboutCard/AboutCard.tsx

```diff
@@ -61,16 +61,20 @@ type CodeLinkInfo = {
 };
 
 function getCodeLinkInfo(entity: Entity): CodeLinkInfo {
+  let sourceUrl;
+  if (entity.metadata?.annotations?.['backstage.io/browser-source-url']) {
+    sourceUrl = entity.metadata.annotations['backstage.io/browser-source-url'];
+  }
+
   const location = findLocationForEntityMeta(entity?.metadata);
   if (location) {
+    sourceUrl = sourceUrl || location.target;
     const type =
-      location.type === 'url'
-        ? determineUrlType(location.target)
-        : location.type;
+      location.type === 'url' ? determineUrlType(sourceUrl) : location.type;
     return {
       icon: iconMap[type],
-      edithref: createEditLink(location),
-      href: location.target,
+      edithref: findEditUrl(entity, location),
+      href: sourceUrl,
     };
   }
```

Comment:
Feel like there's some tweaks to the logic needed here.  For example, even if there's a source URL annotation, if there is no location, this will return blank?  I don't recall, will location ALWAYS be valid?  if so, maybe the if source work should be moved in there, OR maybe this should be a broader with psuedocode like:

```
if sourceUrl annotation has been specified
  use it for source link
else
  try to use the location target for source link
end
return source link
```

Not sure if that makes sense 😁

Reply:
The complicating factor is the edithref which depends on the location, but actually I think that can be dealt with.